### ISSUE TITLE
[a11y]: API Key Modal, Same Heading Text One after the Other

### ISSUE THREAD

**Issue by anamikaanu96** [carbon-design-system/ibm-products#6251]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

v2.52.0-rc.0

### Automated testing tool and ruleset

Manual Testing

### Assistive technology

JAWS

### Description

1. Same Heading Text One after the Other  -Unable to  understand the context, and may confuse  the Screen reader users.
2. Heading level  Hierarchy is not maintained - H1 used for the  Model - which is not recommended.

Expected behaviour :
1. Please confirm the purpose of presenting  the same Heading Text One after the Other , if the same is not  necessary avoid this .
2. As Model is only a subsidiary content of the parent page- it is recommended to use H2- for the Main Heading of the Modal , instead of using H1. 

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-generating-an-api-key-apikeymodal--generate&globals=viewport:basic

### Steps to reproduce

This is reported in API Key Modal - Generate, Custom Generate.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
